exporter/exporterhelper/request/request.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package request // import "go.opentelemetry.io/collector/exporter/exporterhelper/request"
+
+import (
+	"context"
+)
+
+// Request represents a single request that can be sent to an external endpoint.
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+type Request interface {
```

Comment:
For the moment can we keep this internal, since the usage is still internal anyway.

Reply:
Maybe keep it in internal for the moment.

Reply:
The usage is not only internal. This is a public interface used by the new experimental exporterhelper. We are  moving to another place

Reply:
Let't do that in a separate PR then?

Reply:
I don't think we need it in a separate package, that's why I don't want to have it.

Reply:
In this PR both public and internal packages need `Request` and other related API. If we leave it in the `exporterhelper` package, we could use `any` in the `internal` package instead of `Request`, but we also need to translate marshal/unmarshal funcs. Is that what you think of?

I like a separate package for `Request` because it makes the naming a bit nicer. Instead of `exporterhelper.RequestUnmarshaler` we get `request.Unmarshaler`

Reply:
I think we already had everywhere using the `internal.Request`, and I am confused why we had to change too many things?

Reply:
What I'm suggesting in this PR is to stop using the `internal.Request` and use the original request in the senders along with the context. I think this change significantly simplifies code. The only place where we need to wrap it is the queue/internal package.

Reply:
What I am saying is that I don't believe long term we want request in a separate package, so better not to put it there now.

Reply:
Keep in mind that go is duck typing, which means for the moment you can define the interface twice one in exporter and one in internal and everything works.

Reply:
Ok, I moved the Request back to the exporterhelper package